packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,278 @@
+            self.module.fail_json(msg="Failed to create repo.", output=results[1])
+        else:
+            return True
+
+    def delete_repo(self, repo_id):
+        cmd = self.pulp_command
+        cmd += " delete "
+        cmd += " --repo-id=%s " % pipes.quote(repo_id)
+        results = self.module.run_command(cmd)
+        if results[0] != 0:
+            self.module.fail_json(msg="Failed to delete repo.", output=results[1])
+        else:
+            return True
+
+    def get_pulp_absolute_path(self):
+        self.pulp_absolute_path = self.module.get_bin_path('pulp-admin')
+        if not self.pulp_absolute_path:
+            module.fail_json(msg="pulp-admin binary could not be found.")
+
+    def parse_repo_list(self, repo_list):
+        repo_list_lines = repo_list.split("\n")[4:]
+        all_repos = []
+        repo = {}
+
+        for line in repo_list_lines:
+            # If we have a blank line, we're in between 2 repos
+            if line == '':
+                all_repos.append(repo)
+                repo = {}
+                continue
+            if line.startswith('Id:'):
+                repo['repo_id'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('Display Name:'):
+                repo['display_name'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('Description:'):
+                repo['description'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('    Feed:'):
+                repo['feed_url'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('    Relative URL:'):
+                repo['relative_url'] = line.split(":", 1)[1].strip()
+                continue
+        return all_repos
+
+    def set_pulp_base_command(self):
+        cmd = self.pulp_absolute_path
+        if self.username:
+            cmd += " --username=%s" % pipes.quote(self.username)
+        if self.password:
+            cmd += " --password=%s" % pipes.quote(self.password)
+        cmd += " %s repo" % pipes.quote(self.repo_type)
+        cmd += " "
+        self.pulp_command = cmd
+
+    def set_repo_feed_url(self, repo_id, feed_url):
+        cmd = self.pulp_command
+        cmd += "update "
+        cmd += " --repo-id=%s " % repo_id
+        cmd += " --feed=%s " % feed_url
+        update_output = self.module.run_command(cmd)
+        if update_output[0] != 0:
+            self.module.fail_json(msg="Error setting feed URL on repo.")
+
+    def set_repo_relative_url(self, repo_id, relative_url):
+        cmd = self.pulp_command
+        cmd += "update "
+        cmd += " --repo-id=%s " % repo_id
+        cmd += " --relative-url=%s " % relative_url
+        update_output = self.module.run_command(cmd)
+        if update_output[0] != 0:
+            self.module.fail_json(msg="Error setting relative URL on repo.")
+
+    def set_repo_list(self):
+        cmd = self.pulp_command
```

Comment:
Since that method is parsing output ( in parse_repo_list), you need to make sure that the command is run with C locales. See https://github.com/ansible/ansible-modules-core/pull/3257/files for a example.